src/nodeMatchers.ts

```diff
@@ -7,6 +7,39 @@ export function hasType(...typeNames: string[]): NodeMatcher {
     typeNames.includes(node.type) ? simpleSelectionExtractor(node) : null;
 }
 
+export function childNodeMatcher(
+  getNode: (node: SyntaxNode) => SyntaxNode | null,
```

Comment:
I think I'd rename this to
```suggestion
  getMatchingChildNode: (node: SyntaxNode) => SyntaxNode | null,
```